mailet/base/src/main/java/org/apache/mailet/base/AutomaticallySentMailDetectorImpl.java

```diff
@@ -0,0 +1,124 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.mailet.base;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+import javax.mail.MessagingException;
+
+import org.apache.james.mime4j.MimeException;
+import org.apache.james.mime4j.parser.AbstractContentHandler;
+import org.apache.james.mime4j.parser.MimeStreamParser;
+import org.apache.james.mime4j.stream.BodyDescriptor;
+import org.apache.james.mime4j.stream.MimeConfig;
+import org.apache.mailet.Mail;
+
+public class AutomaticallySentMailDetectorImpl implements AutomaticallySentMailDetector {
+
+    public boolean isAutomaticallySent(Mail mail) throws MessagingException {
+        return isMailingList(mail) ||
+            isAutoSubmitted(mail) ||
+            isMdnSentAutomatically(mail);
+    }
+
+    public boolean isMailingList(Mail mail) throws MessagingException {
+        String localPart = mail.getSender().getLocalPart();
```

Comment:
From the slack discussion we had on slack, you confuse 2 concepts : 
 - FROM of the envelope. This is the from of mail.getSender(). Enforced by the SMTP protocol.
 - FROM of the`Mime Message. No conditions on this one, so ie spammer can choose to not specify it.

FROM envellop is enforced by : 
 - SMTP server
 - JMAP server
 - JamesMailetContext and Abstract redirect do not enforce sent mail with non null senders, but with input mails with non null Sender address will end up with non null sender.

I added a Preconditions.checkNotNull